### PR TITLE
Add FormBody::getFields()

### PR DESCRIPTION
**Pull request by trowski**
Needed for generating OAuth 1.0 signatures.

### CODE REVIEW

File: lib/FormBody.php [hunk 1]
```diff
@@ -175,16 +190,20 @@ private function determineContentType() {
     }
 
     public function getBodyLength(): Promise {
+        if ($this->cachedLength) {
+            return $this->cachedLength;
+        }
```
Comment (kelunik) [inline lib/FormBody.php]: This needs to be reset somewhere.

Comment (trowski) [reply lib/FormBody.php]: These were reset in `resetCache()`, but never used anywhere. Leftover from refactoring?

Comment (kelunik) [reply lib/FormBody.php]: Maybe.

**Comment by trowski** [comment]
Good to merge then?

**Pull request merged**